src/desktop/components/search_bar/view.coffee

```diff
@@ -118,13 +118,30 @@ module.exports = class SearchBarView extends Backbone.View
 
   # Display spotlight result if:
   # - more than 4 characters have been typed AND
-  # - query is an exact case-insensitive leading match of the search result AND
-  # - query is more than 30% of the length of the search result
+  # - query is more than 30% of the length of the search result AND
+  # - query is a case-insensitive leading match of the search result or any token
   shouldShowSpotlightSearch: (searchResult) ->
+    @isSubsequentMatch = false
     currentQuery = @$input.val()
     currentQuery.length > 4 &&
-      searchResult.toLowerCase().indexOf(currentQuery.toLowerCase()) is 0 &&
-      (currentQuery.length / searchResult.length) >= 0.3
+      (currentQuery.length / searchResult.length) >= 0.3 &&
+      @isLeadingMatchOfAnyToken(currentQuery, searchResult)
+
+  # Return true if:
+  #  - result starts with word OR
+  #  - any token in result starts with word (sets isSubsequentMatch to true)
+  isLeadingMatchOfAnyToken: (word, result) ->
+    return true if result.toLowerCase().indexOf(word.toLowerCase()) is 0
+
+    if word.indexOf(' ') > -1
+      regex = new RegExp(word, 'i')
+      return @isSubsequentMatch = true if result.match(regex)
+
+    _.each result.split(' '), (token) =>
```

Comment:
Here we do the 'leading edge match' on each token in the search result (to be able to match "warhol" -> "andy warhol").

Worth noting that this method wouldn't even be evaluated in the case of something like "warhol" -> "andy the coolest warhol", (although it would return true) b/c the search term "warhol" is not long enough relative to the search result, so would be caught by the earlier checks.

Reply:
i see - that's fine i think